### ISSUE TITLE
Missing (!a || !b) in _equals

### ISSUE THREAD

**Issue by uniphil** [uniphil/results#83]
[_equals](https://github.com/uniphil/results/blob/master/index.js#L94) should have [this check](https://github.com/facebook/immutable-js/blob/master/src/is.js#L68-L70), that's currently missing.

I think I might have left it out because `a` should usually be an instance of a Union, but that assumption is false.

I don't have a failing test case yet, but here's a stack trace from a real piece of software:

```
/home/ubuntu/[redacted]/node_modules/results/index.js:121
  if (typeof a.valueOf === 'function' && typeof b.valueOf === 'function') {
              ^

TypeError: Cannot read property 'valueOf' of undefined
    at _equals (/home/ubuntu/[redacted]/node_modules/results/index.js:121:15)
    at /home/ubuntu/[redacted]/node_modules/results/index.js:116:14
    at Array.every (native)
    at _equals (/home/ubuntu/[redacted]/node_modules/results/index.js:115:19)
    at UnionOption.equalsProto [as equals] (/home/ubuntu/[redacted]/node_modules/results/index.js:138:10)
    at is (/home/ubuntu/[redacted]/node_modules/immutable/dist/immutable.js:716:16)
    at /home/ubuntu/[redacted]/node_modules/immutable/dist/immutable.js:3996:50
    at /home/ubuntu/[redacted]/node_modules/immutable/dist/immutable.js:1940:16
    at ArrayMapNode.iterate.HashCollisionNode.iterate [as iterate] (/home/ubuntu/[redacted]/node_modules/immutable/dist/immutable.js:2289:11)
    at src_Map__Map.__iterate (/home/ubuntu/[redacted]/node_modules/immutable/dist/immutable.js:1938:32)
```

I don't think a test-case requires interaction with Immutable.js, but it might.

**Comment by uniphil** [comment]
test case:

```
> const results = require('results')
undefined
> U = results.Union({A: null})
{ OptionClass: { [Function: UnionOption] unionFactory: { [Function: Union] is: [Function: _equals] } },
  toString: [Function: toString],
  match: [Function: match],
  A: [Function] }
> a = U.A(undefined)
UnionOption { options: { A: null }, name: 'A', data: [ undefined ] }
> a2 = U.A('')
UnionOption { options: { A: null }, name: 'A', data: [ '' ] }
> a.equals(a2)
TypeError: Cannot read property 'valueOf' of undefined
    at _equals (/home/phil/code/[redacted]/node_modules/results/index.js:124:15)
    at /home/phil/code/[redacted]/node_modules/results/index.js:119:14
    at Array.every (native)
    at _equals (/home/phil/code/[redacted]/node_modules/results/index.js:118:19)
    at UnionOption.equalsProto [as equals] (/home/phil/code/[redacted]/node_modules/results/index.js:141:10)
    at repl:1:3
    at REPLServer.defaultEval (repl.js:260:27)
    at bound (domain.js:287:14)
    at REPLServer.runBound [as eval] (domain.js:300:12)
    at REPLServer.<anonymous> (repl.js:429:12)
```

**Comment by uniphil** [comment]
``` js
U.A('').equals(U.A(undefined))
```

will do it too

**Comment by uniphil** [comment]
incidentally fixed to pass the test suite for 4bab3e931003e6ee80549fd08d937765c62bfe95
